### PR TITLE
Combine DB container tests into single module, and improve error handling/display

### PR DESCRIPTION
**Pull request by rnorth**
~This PR was created on top of #242, and should be merged after that.~

This change includes what was previously covered by #242: Refactoring of all DB container tests into a single separate module. The idea is that this will make it easier to reduce duplication between tests in the future. Along the way, some reliability issues with DB containers have been worked on:
* By default, MySQL containers now use a configuration ini file that reduces initial memory consumption - important in memory constrained environments. Refs #209
* To allow this to work, inclusion of classpath resources that are held inside a JAR has been fixed. Also see #213 
* DB container startup timeout (waiting for test query to return successfully) has been increased from 60s to 120s.

This also goes through a few places where unnecessary stack traces or errors were indicated during tests. It emerged that some behaviour (container reaper and exec-based liveness checks) were running against stopped or deleted containers, causing loud error logging from docker-java.

### CODE REVIEW

File: core/src/main/java/org/testcontainers/containers/GenericContainer.java [hunk 1]
```diff
@@ -815,6 +815,10 @@ public ExecResult execInContainer(Charset outputCharset, String... command)
 
         }
 
+        if (!isRunning()) {
```
Comment (bsideup) [inline core/src/main/java/org/testcontainers/containers/GenericContainer.java]: nice catch!

File: core/src/main/java/org/testcontainers/utility/ResourceReaper.java [hunk 2]
```diff
@@ -67,6 +68,8 @@ public void registerContainerForCleanup(String containerId, String imageName) {
      */
     public void stopAndRemoveContainer(String containerId) {
         stopContainer(containerId, registeredContainers.get(containerId));
+
+        registeredContainers.remove(containerId);
```
Comment (bsideup) [inline core/src/main/java/org/testcontainers/utility/ResourceReaper.java]: maybe we should do it in `finally {}` block?

File: core/src/main/java/org/testcontainers/utility/ResourceReaper.java [hunk 3]
```diff
@@ -83,6 +86,13 @@ public void stopAndRemoveContainer(String containerId, String imageName) {
 
     private void stopContainer(String containerId, String imageName) {
 
+        List<Container> allContainers = dockerClient.listContainersCmd().withShowAll(true).exec();
+
+        if (allContainers.stream().noneMatch(it -> it.getId().equals(containerId))) {
```
Comment (bsideup) [inline core/src/main/java/org/testcontainers/utility/ResourceReaper.java]: just FYI:

```
allContainers.stream().map(Container::getId).noneMatch(containerId::equals)
```

File: core/src/main/java/org/testcontainers/utility/ResourceReaper.java [hunk 4]
```diff
@@ -106,6 +116,13 @@ private void stopContainer(String containerId, String imageName) {
         }
 
         try {
+            InspectContainerResponse containerInfo = dockerClient.inspectContainerCmd(containerId).exec();
```
Comment (bsideup) [inline core/src/main/java/org/testcontainers/utility/ResourceReaper.java]: we don't use `containerInfo` anywhere

File: core/src/main/java/org/testcontainers/utility/ResourceReaper.java (hunk 2 above)
Comment (rnorth) [inline core/src/main/java/org/testcontainers/utility/ResourceReaper.java]: Actually even though it's a little brittle I'd prefer to keep it that way. If `stopContainer` fails for some reason, my feeling is that it's slightly better to _potentially_ try again by leaving the containerId in the registered set.

File: core/src/main/java/org/testcontainers/utility/ResourceReaper.java (hunk 3 above)
Comment (rnorth) [inline core/src/main/java/org/testcontainers/utility/ResourceReaper.java]: Nice 😁

File: core/src/main/java/org/testcontainers/utility/ResourceReaper.java (hunk 4 above)
Comment (rnorth) [inline core/src/main/java/org/testcontainers/utility/ResourceReaper.java]: Yep, will remove

**Pull request merged**